jax/experimental/maps.py

```diff
@@ -668,3 +669,27 @@ def subst_eqn_axis_names(eqn, axis_subst: Dict[AxisName, Tuple[AxisName]]):
     axis_names = (axis_names,)
   new_axis_names = sum((axis_subst.get(name, (name,)) for name in axis_names), ())
   return eqn._replace(params=dict(eqn.params, axis_name=new_axis_names))
+
+
+# -------- soft_pmap --------
+
+def soft_pmap(fun: Callable, axis_name: Optional[AxisName] = None, in_axes=0
```

Comment:
If we're deleting `soft_pmap` from API (which is a breaking change), why don't we remove it completely? Is it because there are still some users internally and you'll need to fix their code?

Reply:
Good point. Yes, the motivation was to keep downstream users working without having to fix their code, but (1) there are currently only two users (though I remember seeing a few more before so that's suspicious), and (2) we could just add this `xmap` wrapper to _their_ code rather than keeping it in ours.

How about this: I'll follow up with fixing the downstream users and deleting `soft_pmap` from the API entirely. But by landing this PR first we get to remove almost all the `soft_pmap` implementation without having to block on those downstream changes.